libs/header/sync/sync.go

```diff
@@ -16,51 +15,45 @@ var log = logging.Logger("header/sync")
 
 // Syncer implements efficient synchronization for headers.
 //
-// Subjective header - the latest known header that is not expired (within trusting period)
-// Network header - the latest header received from the network
+// Subjective Head - the latest known local valid header and a sync target.
+// Network Head - the latest valid network-wide header. Becomes subjective once applied locally.
 //
 // There are two main processes running in Syncer:
-// 1. Main syncing loop(s.syncLoop)
-//   - Performs syncing from the subjective(local chain view) header up to the latest known trusted header
+// - Main syncing loop(s.syncLoop)
+//   - Performs syncing from the latest stored header up to the latest known Subjective Head
 //   - Syncs by requesting missing headers from Exchange or
-//   - By accessing cache of pending and verified headers
+//   - By accessing cache of pending headers
 //
-// 2. Receives new headers from PubSub subnetwork (s.processIncoming)
-//   - Usually, a new header is adjacent to the trusted head and if so, it is simply appended to the local store,
-//     incrementing the subjective height and making it the new latest known trusted header.
-//   - Or, if it receives a header further in the future,
-//     verifies against the latest known trusted header
-//     adds the header to pending cache(making it the latest known trusted header)
-//     and triggers syncing loop to catch up to that point.
+// - Receives every new Network Head from PubSub gossip subnetwork (s.incomingNetworkHead)
+//   - Validates against the latest known Subjective Head, is so
+//   - Sets as the new Subjective Head, which
+//   - if there is a gap between the previous and the new Subjective Head
+//   - Triggers s.syncLoop and saves the Subjective Head in the pending so s.syncLoop can access it
 type Syncer[H header.Header] struct {
-	sub      header.Subscriber[H]
-	exchange header.Exchange[H]
-	store    header.Store[H]
+	sub     header.Subscriber[H]
+	store   syncStore[H]
+	getter  syncGetter[H]
```

Comment:
could you please mention that this getter is fetching the head from the network?

Reply:
Hmm, we fetch everything from it, not only head

Reply:
Yes, right. I meant, that it can be confusing now to understand what is getter, what, and where it gets headers.

We used `header.Exchange`, that has a comment:

```
Exchange encompasses the behavior necessary to request Headers
from the network.
```

Reply:
Ok, added